tests/system-tests/rdscore/internal/rdscorecommon/sriov-pod-level-bond.go

```diff
@@ -0,0 +1,1114 @@
+	scheduleOnHost,
+	sriovNet1Name,
+	sriovNet2Name,
+	bondNetName,
+	bondInfIPv4,
+	bondInfIPv6,
+	bondInfSubMaskIPv4,
+	bondInfSubMaskIPv6,
+	bondInfMacAddr string,
+	deployLabels map[string]string) *deployment.Builder {
+	glog.V(100).Infof("Defining deployment %q in %q ns", deploymentName, nsName)
+
+	nodeSelector := map[string]string{"kubernetes.io/hostname": scheduleOnHost}
+
+	netAnnotations := []*types.NetworkSelectionElement{
+		{
+			Name:      sriovNet1Name,
+			Namespace: nsName,
+		},
+		{
+			Name:      sriovNet2Name,
+			Namespace: nsName,
+		},
+		{
+			Name:      bondNetName,
+			Namespace: nsName,
+			IPRequest: []string{fmt.Sprintf("%s/%s", bondInfIPv4, bondInfSubMaskIPv4),
+				fmt.Sprintf("%s/%s", bondInfIPv6, bondInfSubMaskIPv6)},
+			MacRequest: bondInfMacAddr,
+		},
+	}
+
+	podDeployment := deployment.NewBuilder(apiClient, deploymentName, nsName, deployLabels, *containerConfig)
+
+	glog.V(100).Infof("Assigning ServiceAccount %q to the deployment", podLevelBondDeploymentSAName)
+
+	podDeployment = podDeployment.WithServiceAccountName(podLevelBondDeploymentSAName)
+
+	podDeployment = podDeployment.WithNodeSelector(nodeSelector)
+
+	podDeployment = podDeployment.WithSecondaryNetwork(netAnnotations)
+
+	return podDeployment
+}
+
+func generateTCPTraffic(clientPod *pod.Builder, serverIPAddr, serverPort string) (string, error) {
+	By("Generating TCP traffic")
+
+	glog.V(100).Infof("Ensure pod %q in namespace %q is Ready",
+		clientPod.Definition.Name, clientPod.Definition.Namespace)
+
+	err := clientPod.WaitUntilReady(5 * time.Second)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to wait for pod %q in namespace %q to become Ready: %v",
+			clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+
+		return "", fmt.Errorf("failed to wait for pod %q in namespace %q to become Ready: %w",
+			clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+	}
+
+	cmdToRun := []string{"bash", "-c",
+		fmt.Sprintf("testcmd -protocol tcp -port %s -interface net3 -server %s",
+			serverPort, serverIPAddr)}
+
+	glog.V(100).Infof("Execute command: %q", cmdToRun)
+
+	var output string
+
+	err = wait.PollUntilContextTimeout(
+		context.TODO(),
+		time.Second*5,
+		time.Minute*1,
+		true,
+		func(ctx context.Context) (bool, error) {
+			result, err := clientPod.ExecCommand(cmdToRun, clientPod.Object.Spec.Containers[0].Name)
+
+			if err != nil {
+				glog.V(100).Infof("Error running command from within a pod %q: %v",
+					clientPod.Object.Name, err)
+
+				return false, nil
+			}
+
+			glog.V(100).Infof("Successfully executed command from within a pod %q in namespace %q",
+				clientPod.Object.Name, clientPod.Definition.Namespace)
+
+			output = result.String()
+			glog.V(100).Infof("Command's output:\n\t%v", output)
+
+			return true, nil
+		})
+
+	if err != nil {
+		return "", fmt.Errorf("failed to run command from within pod %s: %w", clientPod.Object.Name, err)
+	}
+
+	return output, nil
+}
+
+func generateLongTCPTraffic(
```

Comment:
The only difference from `generateTCPTraffic` I spotted are params of the _cmdToRun_ variable.
Does it make sense to add extra params to the aforementioned function. to avoid code duplication?